sw/host/opentitantool/src/command/spx.rs

```diff
@@ -0,0 +1,171 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use serde_annotate::Annotate;
+use std::any::Any;
+use std::path::{Path, PathBuf};
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::crypto::spx::{SpxKeypair, SpxPublicKey, SpxPublicKeyPart, SpxSignature};
+use opentitanlib::image::image::Image;
+use opentitanlib::util::file::{FromReader, PemSerilizable, ToWriter};
+
+/// Given the path to a public key, returns the public key. Given
+/// the path to a full keypair, extracts the public key from the private
+/// key and returns the public key.
+fn load_pub_or_priv_key(path: &Path) -> Result<SpxPublicKey> {
```

Comment:
Would it be worthwhile to build this functionality into `SpxPublicKey` rather than having a separate utility function?